diff --git a/gdal.cmake b/gdal.cmake
index b5e5e4f..d8484cd 100644
--- a/gdal.cmake
+++ b/gdal.cmake
@@ -611,9 +611,11 @@ install(
   FRAMEWORK DESTINATION "${FRAMEWORK_DESTINATION}")
 
 # Generate targets file for importing directly from GDAL build tree
-export(TARGETS ${GDAL_LIB_TARGET_NAME}
-        NAMESPACE GDAL::
-        FILE "GDAL-targets.cmake")
+if(NOT CMAKE_SKIP_INSTALL_RULES)
+  export(TARGETS ${GDAL_LIB_TARGET_NAME}
+          NAMESPACE GDAL::
+          FILE "GDAL-targets.cmake")
+endif()
 
 if (NOT GDAL_ENABLE_MACOSX_FRAMEWORK)
   # Generate GdalConfig.cmake and GdalConfigVersion.cmake
