--- a/src/base_station.h
+++ b/src/base_station.h
@@ -107,6 +107,6 @@
 			FixedPositionSettings fixed_position;
 		} settings;
 	};
-	BaseSettings _base_settings;
+	BaseSettings _base_settings{};
 };

--- a/src/ashtech.cpp
+++ b/src/ashtech.cpp
@@ -649,7 +649,7 @@
 			ASH_DEBUG("Connected board: %i", (int)_board);
 		}

-	} else if (memcmp(_rx_buffer, "$PASHR,RECEIPT,", 15) == 0) {
+	} else if (_output_mode == OutputMode::RTCM && memcmp(_rx_buffer, "$PASHR,RECEIPT,", 15) == 0) {
 		// this is the response to $PASHS,POS,AVG,100
 		// example: $PASHR,RECEIPT,POS,AVG,STARTED,INTERVAL,100,114502.56,28.12.2011
 		if (_command_state == NMEACommandState::waiting && _waiting_for_command == NMEACommand::RECEIPT) {
@@ -710,7 +710,7 @@


 	// handle survey-in status update
-	if (_survey_in_start != 0) {
+	if (_output_mode == OutputMode::RTCM && _survey_in_start != 0) {
 		const gps_abstime now = gps_absolute_time();
 		uint32_t survey_in_duration = (now - _survey_in_start) / 1000000;

@@ -1231,7 +1231,8 @@
 	status.latitude = latitude;
 	status.longitude = longitude;
 	status.altitude = altitude;
-	status.duration = _base_settings.settings.survey_in.min_dur;
+	status.duration = _base_settings.type == BaseSettingsType::survey_in ?
+			  _base_settings.settings.survey_in.min_dur : 0;
 	status.mean_accuracy = 0; // unknown
 	status.flags = (int)valid | ((int)active << 1);
 	surveyInStatus(status);
--- a/src/femtomes.cpp
+++ b/src/femtomes.cpp
@@ -624,6 +624,7 @@

 		} else {
 			FEMTO_ERR("Femto: command LOG UAVGPSB 0.1 failed");
+			return -1;
 		}

 		if (_satellite_info) {
@@ -728,7 +729,8 @@
 	status.latitude = latitude;
 	status.longitude = longitude;
 	status.altitude = altitude;
-	status.duration = _base_settings.settings.survey_in.min_dur;
+	status.duration = _base_settings.type == BaseSettingsType::survey_in ?
+			  _base_settings.settings.survey_in.min_dur : 0;
 	status.mean_accuracy = 0; // unknown
 	status.flags = (int)valid | ((int)active << 1);
 	surveyInStatus(status);
