diff --git a/src/GPS/CMakeLists.txt b/src/GPS/CMakeLists.txt
index 4b75ad46c..3b1240f81 100644
--- a/src/GPS/CMakeLists.txt
+++ b/src/GPS/CMakeLists.txt
@@ -28,7 +28,7 @@ target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_
 CPMAddPackage(
     NAME px4-gpsdrivers
     GITHUB_REPOSITORY PX4/PX4-GPSDrivers
-    GIT_TAG main
+    GIT_TAG 0b9695881bd1e8f830ab4538ab3acc0050019eba
     SOURCE_SUBDIR src
 )
 
