--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,14 +30,22 @@ set(EXAMPLE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/${MAVLINK_DIALECT}/mavlink.h)
 add_custom_command(OUTPUT ${EXAMPLE_HEADER}
-    COMMAND ${Python_EXECUTABLE}
+    COMMAND "${Python_EXECUTABLE}"
-        -m pip install -r pymavlink/requirements.txt --upgrade -t ${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/
-        COMMAND ${CMAKE_COMMAND} -E env "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/" ${Python_EXECUTABLE}
+        "${CMAKE_SOURCE_DIR}/tools/setup/run_with_retry.py"
+        --
+        "${Python_EXECUTABLE}"
+        -m pip install -r pymavlink/requirements.txt --upgrade
+        -t "${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/"
+    COMMAND "${CMAKE_COMMAND}" -E env "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/" "PYTHONHASHSEED=0" "${Python_EXECUTABLE}"
         -m pymavlink.tools.mavgen
         --lang=C
         --wire-protocol=${MAVLINK_VERSION}
         --output ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/
         message_definitions/v1.0/${MAVLINK_DIALECT}.xml
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    DEPENDS message_definitions/v1.0/${MAVLINK_DIALECT}.xml
-    COMMENT "Generating C headers")
+    DEPENDS
+        message_definitions/v1.0/${MAVLINK_DIALECT}.xml
+        "${CMAKE_SOURCE_DIR}/tools/setup/run_with_retry.py"
+        "${CMAKE_SOURCE_DIR}/tools/common/proc.py"
+    COMMENT "Generating C headers"
+    VERBATIM)
-
+
 # Unfortunately, the dependencies don't work for INTERFACE libraries.
