--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,9 +43,13 @@
 endif()

 add_custom_command(OUTPUT ${EXAMPLE_HEADER}
-    COMMAND ${Python_EXECUTABLE}
-        -m pip install -r pymavlink/requirements.txt --upgrade -t ${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/
-        COMMAND ${CMAKE_COMMAND} -E env "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/" ${Python_EXECUTABLE}
+    COMMAND "${Python_EXECUTABLE}"
+        "${CMAKE_SOURCE_DIR}/tools/setup/run_with_retry.py"
+        --
+        "${Python_EXECUTABLE}"
+        -m pip install -r pymavlink/requirements.txt --upgrade
+        -t "${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/"
+    COMMAND "${CMAKE_COMMAND}" -E env "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/" "PYTHONHASHSEED=0" "${Python_EXECUTABLE}"
         -m pymavlink.tools.mavgen
         --lang=C
         --wire-protocol=${MAVLINK_VERSION}
@@ -53,8 +57,12 @@
         message_definitions/v1.0/${MAVLINK_DIALECT}.xml
         ${GENERATE_CXX_COMMAND}
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    DEPENDS message_definitions/v1.0/${MAVLINK_DIALECT}.xml
-    COMMENT "Generating C and C++11 headers")
+    DEPENDS
+        message_definitions/v1.0/${MAVLINK_DIALECT}.xml
+        "${CMAKE_SOURCE_DIR}/tools/setup/run_with_retry.py"
+        "${CMAKE_SOURCE_DIR}/tools/common/proc.py"
+    COMMENT "Generating C headers"
+    VERBATIM)

 # Unfortunately, the dependencies don't work for INTERFACE libraries.
 # The only way I could make it work is to add ALL which means it
--- a/pymavlink/generator/mavgen_c.py
+++ b/pymavlink/generator/mavgen_c.py
@@ -24,6 +24,5 @@ def generate_version_h(directory, xml):
 #ifndef MAVLINK_VERSION_H
 #define MAVLINK_VERSION_H

-#define MAVLINK_BUILD_DATE "${parse_time}"
 #define MAVLINK_WIRE_PROTOCOL_VERSION "${wire_protocol_version}"
 #define MAVLINK_MAX_DIALECT_PAYLOAD_SIZE ${largest_payload}
